pallets/gear/src/lib.rs

```diff
@@ -822,8 +820,8 @@ pub mod pallet {
         }
 
         /// Returns true if `program_id` is that of a in active status or the builtin actor.
-        pub fn is_active(program_id: ProgramId) -> bool {
-            T::BuiltinCache::exists(&program_id)
```

Comment:
@grishasobol here and in `program_exists` too

Reply:
A ok see now lazy initialisation. Was it a bug ? Because in case of builtins list update this cache will stay not updated.

Actually I don't like refactoring your have made)) but agree with you that this cache must be updated each block maybe in on_initialize()